File: frontend/src/api/images.ts
import { api } from './client';

export interface PresignResponse {
  upload_url: string;
  public_url: string;
  s3_key: string;
  expires_in: number;
}

export interface ImageResponse {
  image_id: number;
  user_id: number;
  storage_url: string;
  s3_key: string;
  uploaded_at: string;
}

export const imagesApi = {
  presign: (body: { file_name: string; mime_type: string; file_size: number }) =>
    api.post<PresignResponse>('/files/presign', body),

  createImage: (body: {
    storage_url: string;
    s3_key: string;
    original_file_name: string;
    mime_type: string;
    file_size: number;
    crop_data: { x: number; y: number; width: number; height: number };
    upload_status: string;
  }) => api.post<ImageResponse>('/images', body),

  /** S3 presigned URL로 blob 직접 PUT (백엔드 미경유) */
  uploadToS3: async (uploadUrl: string, blob: Blob): Promise<void> => {
    await fetch(uploadUrl, {
      method: 'PUT',
      body: blob,
      headers: { 'Content-Type': 'image/jpeg' },
    });
  },
};

File: backend/app/schemas/images.py
from pydantic import BaseModel, Field


class ImageCreateRequest(BaseModel):
    storage_url: str = Field(min_length=1, max_length=1000)
    s3_key: str = Field(min_length=1, max_length=500)
    original_file_name: str | None = Field(default=None, max_length=255)
    mime_type: str | None = Field(default=None, max_length=100)
    file_size: int | None = Field(default=None, ge=1)
    crop_data: dict | None = None
    upload_status: str = Field(default="UPLOADED", max_length=30)


class ImageCreateResponse(BaseModel):
    image_id: int
    user_id: int
    storage_url: str
    s3_key: str
    uploaded_at: str
